lib/options_validator.js

```diff
@@ -7,6 +7,43 @@ const VALIDATION_LEVEL_NONE = 'none';
 const VALIDATION_LEVEL_WARN = 'warn';
 const VALIDATION_LEVEL_ERROR = 'error';
 
+/**
+ * Checks that arguments provided to an operation match that operation's arity.
+ * An error will be thrown if the operation is called with an incorrect number of arguments.
+ *
+ * @method
+ * @param {object} args Arguments supplied to the operation that is calling assertArity.
+ * @param {number} requiredArity The correct arity of the operation.
+ */
+function assertArity(args, requiredArity) {
+  // acceptable values for arity are 0, 1, and 2
+  if (requiredArity < 0 || requiredArity > 2) {
+    throw new Error('The arity of an operation can only be 0, 1, or 2.');
+  }
+
+  // requiredArity + 2 is the maximum number of arguments allowed, since this accounts for options and callback
+  // requiredArity is the minimum number of arguments allowed
+  if (args.length > requiredArity + 2 || args.length < requiredArity) {
+    const invalidateMessageArity = `This operation has a required arity of ${requiredArity}, but ${
+      args.length
+    } arguments were provided.`;
+    invalidate(VALIDATION_LEVEL_ERROR, invalidateMessageArity);
+  }
+
+  let numberOfArguments = args.length;
+
+  // a callback does not count towards the number of arguments
+  if (typeof args[args.length - 1] === 'function') {
+    numberOfArguments = numberOfArguments - 1;
+  }
+
+  // we are no longer considering the callback so requiredArity + 1 is the maximum number of arguments allowed
+  if (numberOfArguments > requiredArity + 1) {
```

Comment:
Why do we need to do this check again?

Reply:
This is checking that one of the optional arguments provided was a callback. If you think I should be strictly checking the arity and not verifying the arguments, I can remove this. I guess that this is a little out of scope of just asserting the arity.

Reply:
Ah, so we're changing it later. Got it.

Reply:
That makes sense. Is it possible for both messages to go off at the same time?

Reply:
I don't think so, but it does make more sense for this check to be with the first check so I combined them.